.github/workflows/ci.yml

```diff
@@ -0,0 +1,51 @@
+name: CI
+
+on: [push]
+
+jobs:
+  build:
+    name: ${{ matrix.config.name }}
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      matrix:
+        config:
+          - {
+            name: linux,
```

Comment:
Supported OS are listed here https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners
For Windows and MacOS there is only 1 OS available, which is quite recent. I don't known if it means that lib won't work on more recent system.
For Linux we can choose between ubuntu 16 and 18. Would it be interesting to choose ubuntu 16 to increase compatibity with other (often old) OS?

Reply:
I think we should try to validate on different OS version. I propose to test on Ubuntu 16 and 18, centos 6 and 7 (and 8). We do not use hype c++ features so binaries should be compatible but you’re right we have to ensure it works fine.
For MacOS and Windows, should we specify the version, in case several version would be supported to not have unexpected side effects?

Reply:
I did more test, and the version I generated using github actions is not compatible with CentOS 6 (C++ not supported), and CentOS 7 (the glibc is a little bit too old). I think we could ignore the support of centos 6, but we need to support centos 7.
I'm currently trying to compile on Ubuntu using an old version of gcc (4.8), which should build a binary compatible with centos 7, hopefully.